app/src/main/java/co/nimblehq/extension/ErrorExt.kt

```diff
@@ -0,0 +1,20 @@
+package co.nimblehq.extension
+
+import android.content.Context
+import co.nimblehq.R
+import co.nimblehq.domain.data.error.AppError
+import co.nimblehq.domain.data.error.ValidateError
+
+fun Throwable.userReadableMessage(context: Context, vararg formatArgs: Any?): String {
```

Comment:
How differ between "readableMessage" in AppError & "userReadableMessage" in here ,Do these serve the same purpose? if yes, why do we need to create a new extension 🤔

Reply:
`readableMessage` is the message that we get from API.
`userReadableMessage` is the message that we display in UI (if `readableMessage` is `null` or empty we set a general message for it)

Reply:
Hmm 🤔 readableMessage is **inside** AppError and ValidateError **extend** from **AppError** also that why I think we can have only one(**userReadableMessage/readableMessage**) in AppError because we can handler error message for user readable  in VM and then we just send back a message string to view.

val errorGeneralMessage = context.getString(R.string.error_generic)
    return when (throwable) {
        is ValidateError.InvalidEmailError -> context.getString(R.string.error_email_invalid)
        is ValidateError.InvalidPasswordError -> throwable.message ?: errorGeneralMessage
        else -> errorGeneralMessage
    }

Reply:
@Wadeewee ah I see what you mean, we should check the condition for **AppError** instead of **ValidateError**. I'll put a TODO comment here to fix in later PR because it's out of scope. Thanks for pointing the issue out 😄